x-pack/filebeat/input/httpjson/request.go

```diff
@@ -146,38 +164,177 @@ func (rf *requestFactory) newHTTPRequest(stdCtx context.Context, trCtx *transfor
 type requester struct {
 	log               *logp.Logger
 	client            *httpClient
-	requestFactory    *requestFactory
+	requestFactories  []*requestFactory
 	responseProcessor *responseProcessor
 }
 
 func newRequester(
 	client *httpClient,
-	requestFactory *requestFactory,
+	requestFactory []*requestFactory,
 	responseProcessor *responseProcessor,
 	log *logp.Logger) *requester {
 	return &requester{
 		log:               log,
 		client:            client,
-		requestFactory:    requestFactory,
+		requestFactories:  requestFactory,
 		responseProcessor: responseProcessor,
 	}
 }
 
-func (r *requester) doRequest(stdCtx context.Context, trCtx *transformContext, publisher inputcursor.Publisher) error {
-	req, err := r.requestFactory.newHTTPRequest(stdCtx, trCtx)
+// collectResponse returns response from provided request
+func (rf *requestFactory) collectResponse(stdCtx context.Context, trCtx *transformContext, r *requester, publisher inputcursor.Publisher) (*http.Response, error) {
+	req, err := rf.newHTTPRequest(stdCtx, trCtx)
 	if err != nil {
-		return fmt.Errorf("failed to create http request: %w", err)
+		return nil, fmt.Errorf("failed to create http request: %w", err)
 	}
-
 	httpResp, err := r.client.do(stdCtx, trCtx, req)
 	if err != nil {
-		return fmt.Errorf("failed to execute http client.Do: %w", err)
+		return nil, fmt.Errorf("failed to execute http client.Do: %w", err)
+	}
+	return httpResp, nil
+}
+
+// generateNewUrl returns new url value using replacement from oldUrl with ids
+func generateNewUrl(replacement string, oldUrl string, ids []byte) (url.URL, error) {
+	reg, err := regexp.Compile(replacement)
+	if err != nil {
+		return url.URL{}, fmt.Errorf("failed to create regex on provided value: %w", err)
+	}
+	newUrl, err := url.Parse(string(reg.ReplaceAll([]byte(oldUrl), ids)))
```

Comment:
You can use `ReplaceAllString` instead
```suggestion
	newUrl, err := url.Parse(reg.ReplaceAllString(oldUrl, ids))
```

Maybe it also makes sense to have `ids` as a string? So this function has got all types matching and the caller is the responsible for doing type casts, if needed.

Reply:
Got it.
Thanks